### PR TITLE
configurable date renderer

### PR DESCRIPTION
**Pull request by benoitjchevalier**
Hey guys,

was using the date renderer for a benchmark project we have. I realised it was currently hard coded to a specific display and data format.

This PR aims at making the date renderer more flexible by allowing the dev to:
- Choose whether to hide date and/or time in the edit `px-datetime-picker`
- Choose a valid moment.js format from which his data can be interpreted
- Choose a valid moment.js format which will be used to display the data (non edit)

It also uses our internal `window.Px.moment` instance which is the one we rely on for localization (can expand on that if needed but long story short we use this instance instead of the usual `window.moment` one. To localize moment people use our custom methods and localize our custom instance)

the options can be passed in the `columns` definition in a `rendererConfig` object:
```
{
              id: 'timestamp[date]',
              name: 'Timestamp',
              path: 'timestamp',
              type: 'date',
              renderer: 'px-data-grid-date-renderer',
              rendererConfig: {
                dataFormat: 'x',
                displayFormat: 'DD-MM-YYYY HH:mm',
                hideTime
              },
              minBound: 1,
              maxBound: 10,
              hidden: false,
              frozen: false,
              required: false,
              flexGrow: 1
            },
}
```

We'll build a demo with different examples of config later. This PR updates the current custom column demo to show how to configure it.

Let me know if you need more info.

### CODE REVIEW

**Comment by limonte** [comment]
Hello @benoitjchevalier and thank you for the contribution!

Something has happened to https://github.com/vaadin/px-data-grid/pull/526/commits/61af94c1273a15a856ee03ae9d5203899c7bdbfb as it contains rollback changes for miltiple commits which are already in master. Please fix that so we can start reviewing your actual code.  

PS. also, please sign the CLA by clicking on the CLA bot badge.

**Comment by benoitjchevalier** [comment]
Hi @limonte 

I've merged master back into my fork and also fixed the mess I had made in the first place, sorry about that.

I've also made changes to the PR to align with your coding guidelines so that Travis won't fail on the lint task (even though some of those rules don't match the guidelines on our side).

I can't seem to be able to reach the CLA signing, any way we can work around it since we'll eventually get that code back on our org?

Thanks for your time!

**Comment by limonte** [comment]
> I've merged master back into my fork and also fixed the mess I had made in the first place, sorry about that.
> 
> I've also made changes to the PR to align with your coding guidelines so that Travis won't fail on the lint task (even though some of those rules don't match the guidelines on our side).

Perfect, thanks @benoitjchevalier!

> I can't seem to be able to reach the CLA signing, any way we can work around it since we'll eventually get that code back on our org?

Yeah, I guess that's fine.

---

Just one small documentation improvement mentioned by @toshegg before merging :rocket:

**Comment by benoitjchevalier** [comment]
Updated the documentation 📜

**Comment by alump** [comment]
Merging this is delayed for some time, as we are busy with all Q4 issues.

File: px-data-grid.html [hunk 1]
```diff
@@ -325,6 +325,8 @@
              *   * `{string} renderer` - You can define your own renderer element to display and edit cell data in your own way.
              *     For more information, please check sample renderers, e.g. `px-data-grid-number-renderer`, `px-data-grid-date-renderer`.
              *
+             *   * `{object} rendererConfig` - An arbitrary objet that can be used to pass down options to the chosen renderer.
```
Comment (toshegg) [inline px-data-grid.html]: `objet` - typo

**Comment by benoitjchevalier** [comment]
merged and fixed typo

File: px-data-grid-filters-preview-tooltip.html [hunk 2]
```diff
@@ -0,0 +1,177 @@
+<link rel="import" href="../polymer/polymer-element.html">
```
Comment (toshegg) [inline px-data-grid-filters-preview-tooltip.html]: Seems like you merged something wrong, this file has been deleted from px-data-grid.

**Comment by benoitjchevalier** [comment]
Oh yeah, seems like the file hasn't been deleted when merging upstream/master, not entirely sure why. Maybe because I had made some changes to it?

I've removed it now. Getting there!

**Comment by benoitjchevalier** [comment]
I've re-merged master and did not re-generate the api files to avoid conflicts. Didn't want to change the .gitignore file since it's checked in and would lead to other people ignoring the api files as well.

**Comment by alump** [comment]
@toshegg - if you think that this will not cause regression, merge it.

File: bower.json [hunk 3]
```diff
@@ -29,7 +29,8 @@
     "px-actionable-design": "^2.0.5",
     "px-popover": "^2.1.2",
     "px-slider": "^2.1.1",
-    "px-rangepicker": "^3.2.0"
+    "px-rangepicker": "^3.1.0",
```
Comment (alump) [inline bower.json]: This would revert back to old version?

Comment (limonte) [reply bower.json]: No, `^3.1.0`,  and `^3.1.0` are basically the same and mean `3.x.x`.

https://stackoverflow.com/a/22345808/1331425

But yeah, this change is unnecessary.

**Comment by benoitjchevalier** [comment]
Yeah I added a coma after the rangepicker dependency when creating this PR (to add a new dependency below), you guys then updated the rangepicker version and git then decided to keep my "change" rather than merging both when I merged the upstream master. Sigh...

Update the bower dependency (even though it has no consequences) and re-merged master to hopefully avoid daily merge conflict

**Pull request merged**